src/main/java/io/kamax/matrix/json/event/MatrixJsonRoomTagsEvent.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * matrix-java-sdk - Matrix Client SDK for Java
+ * Copyright (C) 2018 Arne Augenstein
+ *
+ * https://www.kamax.io/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.kamax.matrix.json.event;
+
+import com.google.gson.JsonObject;
+
+import io.kamax.matrix.event._RoomTagsEvent;
+import io.kamax.matrix.json.GsonUtil;
+import io.kamax.matrix.json.MatrixJsonObject;
+import io.kamax.matrix.room.RoomTag;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MatrixJsonRoomTagsEvent extends MatrixJsonObject implements _RoomTagsEvent {
+
+    private Content content;
+
+    private String type;
+
+    public MatrixJsonRoomTagsEvent(JsonObject obj) {
+        super(obj);
+        type = getString("type");
+        content = new Content(getObj("content"));
+    }
+
+    @Override
+    public String getType() {
+        return type;
+    }
+
+    private class Content extends MatrixJsonObject {
+
+        private List<RoomTag> tags = new ArrayList<>();
+
+        public Content(JsonObject obj) {
+            super(obj);
+
+            String roomId = getStringOrNull("roomId");
```

Comment:
I don't understand how you can have room tags if you don't have an an associated Room ID

Reply:
According to the spec, the room id is an optional parameter as it is not needed, if the event is part of the JoinedRoom-object of sync. Reference: https://matrix.org/speculator/spec/HEAD/client_server/unstable.html#id457

Reply:
Yes but the class is MatrixJson**Room**TagsEvent

Reply:
I feel like we are not talking about the same thing. Tags are always mapped to a room, but that doesn't necessarily mean the room id has to be part of the event itself. The room can be implicitely specified by the embedding JSON-object, in which case the tag itself does not contain the room id.

Reply:
That's my point: sometimes you can have `null`, sometimes you have a String. It's confusing for anyone using the SDK since they might not know where the object was created, and so they would need to constantly check for it. I think we should never include it, or always include it, but not sometimes.

Reply:
Ah, now I get it. You are totally right. Looking at the class `RoomTag` I just realized that I don't even have a get-method for the room id. So it's totally useless to read it here. I will remove it completely as the user should never need to read it from the tag directly. It should always be clear from the context, to which room the tag belongs.

Reply:
Removed for now. If the real life tests show that users need the room id, I will add it again and make sure it's always set.